content/ConnectingToDatabase/ConnectingToDatabase.q.md

```diff
@@ -0,0 +1,666 @@
+    $password = 'Admin01';
+    
+    $dsn = "pgsql:host=;port=;dbname=;user=;password=";
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected!";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }   
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is its output?
+
+- `An error occurred!`
+
+- No output is displayed.
+
+- `Successfully connected!`
+
+- `Unable to establish a connection.`
+
+- `"Unable to establish a connection."`
+
+
+/// type=MS, answer=[2,3]
+
+Which statements correctly describe the error?
+
+- On line 8, the `$dsn` variable is an invalid variable name.
+
+- Missing variables `$host`, `$db`, `$port`, `$username`, and `$password` on line 8.
+
+- On line 8, the statement `$dsn = "pgsql:host=;port=;dbname=;user=;password=";` is incorrect.
+
+- The statement `$dsn = "pgsql:host=;port=;dbname=;user=;password=";` is written outside the `if` statement.
+
+- There are no parentheses `()` enclosing the statement `$dsn = "pgsql:host=;port=;dbname=;user=;password=";` on line 8.
+
+:::
+
+
+/// type=CR, answer=[tests/ConnectingToDatabase/MissingAssignedVariablesTest.php]
+
+Correct the code so that it outputs the string `Successfully connected!`.
+
+```php
+<?php
+    $host = 'localhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    
+    $dsn = "pgsql:host=;port=;dbname=;user=;password=";
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected!";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }     
+?>
+```
+
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/ConnectingToDatabase/CreatePHPApplicationConnectingToPostgreSQLTest.php]
+
+Write a PHP program that uses the `PDO_PGSQL` driver, `PDO_PGSQL DSN`, and `PDO` class to connect to the PostgreSQL database. First, assign the following variables with its respective values: `$host='localhost'`, `$db = 'MyDatabase'`, `$port = '5432'`, `$username = 'postgres'`, and `$password = 'Admin01'`. Then, assign the `$dsn` variable to the `PDO_PGSQL DSN` statement which contains the DSN parameters `host`, `port`, `dbname`, `username`, and `password`. Assign the DSN parameters with their respective variables `$host`, `$db`, `$port`, `$username`, and `$password`. Next, add a statement that creates the `$conn` object an instance of the `PDO` class which passes the argument `$dsn`. Write the `if` statement to evaluate the `$conn` object inside the parentheses `()`. If the expression evaluates to `true` then, the `echo` statement displays the string `Successfully connected!`. Use the `try` and `catch` statement to evaluate the `if` statement. Set the `catch` block to display the error message `Unable to establish a connection` if there is a problem with the connection. Run the program to view the output.
```

Comment:
Please replace `assign` of this statement `assign the following variables with its respective values:` with `add` since you already specified the statement `$host='localhost'`...

Or you can rephrase the statement to...

First, create the variables `$host`, `$db`, and... and assign the values `localhost`, `MyDatabase`, and... respectively.

Reply:
I think we need to rephrase this statement __Then, assign the `$dsn` variable to the `PDO_PGSQL DSN` statement which contains the DSN parameters `host`, `port`, `dbname`, `username`, and `password`.__

Are we going to `assign the $dsn variable to the `PDO_PGSQL DSN` statement`? or the other way around? Please check. Thanks.

Reply:
Please review this one and rephrase:

> Write the `if` statement to evaluate the `$conn` object inside the parentheses `()`. If the expression evaluates to `true` then, the `echo` statement displays the string `Successfully connected!`. Use the `try` and `catch` statement to evaluate the `if` statement. Set the `catch` block to display the error message `Unable to establish a connection` if there is a problem with the connection.

Remember that we need a `try` here and we need to properly inform the learners of where to place the what command. Please check the rest of the courses on how we create our instructions. Thanks.